PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/common/TextWithErrorHintedField.java

```diff
@@ -0,0 +1,80 @@
+package com.microsoft.azure.sparkserverless.common;
+
+import com.intellij.ui.DocumentAdapter;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.azurecommons.helpers.Nullable;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+
+import javax.swing.*;
+import javax.swing.event.DocumentEvent;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class TextWithErrorHintedField extends JTextField implements Validatable {
+    @Nullable
+    private String errorMessage;
+    @Nullable
+    private Set<String> notAllowedValues;
+    @NotNull
+    private final ErrorMessageTooltip errorMessageTooltip = new ErrorMessageTooltip(this);
+    @NotNull
+    private Pair<Pattern, String> patternAndErrorMessagePair = Pair.of(Pattern.compile("^[a-zA-Z0-9_-]+"),
+            "Only alphanumeric characters, underscores and hyphens are allowed");
+
+    public TextWithErrorHintedField() {
+        super();
+        this.getDocument().addDocumentListener(new DocumentAdapter() {
+            @Override
+            protected void textChanged(DocumentEvent e) {
+                errorMessageTooltip.showForError(self());
+            }
+        });
+    }
+
+    @Nullable
+    protected String validateTextField() {
+        if (StringUtils.isEmpty(getText())) {
+            setErrorMessage("A string is expected");
+        } else if (patternAndErrorMessagePair != null && patternAndErrorMessagePair.getLeft() != null &&
+                !patternAndErrorMessagePair.getLeft().matcher(getText()).matches()) {
+            if (patternAndErrorMessagePair.getRight() != null) {
+                setErrorMessage(patternAndErrorMessagePair.getRight());
+            } else {
+                setErrorMessage(String.format("%s does not match the pattern %s", getText(),
+                        patternAndErrorMessagePair.getLeft().toString()));
+            }
+        } else if (notAllowedValues != null && notAllowedValues.contains(getText())) {
+            setErrorMessage(String.format("%s already exists", getText()));
+        } else {
+            setErrorMessage(null);
+        }
+        return getErrorMessage();
+    }
+
+    @NotNull
+    protected TextWithErrorHintedField self() {
+        return this;
+    }
+
+    @Nullable
+    public String getErrorMessage() {
+        return errorMessage;
+    }
+
+    private void setErrorMessage(@Nullable String errorMessage) {
+        this.errorMessage = errorMessage;
+    }
+
+    public void setStringSet(@Nullable Set<String> notAllowedValues) {
```

Comment:
The method name should be aligned with `notAllowedValues`, such as `setNotAllowedValues`